radio/src/targets/simu/simpgmspace.cpp

```diff
@@ -275,28 +276,19 @@ void audioConsumeCurrentBuffer()
 {
 }
 
-void setScaledVolume(uint8_t volume)
+void audioSetVolume(uint8_t volume)
 {
   simuAudio.currentVolume = 127 * volume * simuAudio.volumeGain / VOLUME_LEVEL_MAX / 10;
   // TRACE_SIMPGMSPACE("setVolume(): in: %u, out: %u", volume, simuAudio.currentVolume);
 }
 
-void setVolume(uint8_t volume)
-{
-}
-
-int32_t getVolume()
-{
-  return 0;
-}
-
 #if defined(SIMU_AUDIO)
-void copyBuffer(uint8_t * dest, const uint16_t * buff, unsigned int samples)
+void copyBuffer(void* dest, const int16_t* buff, unsigned samples)
 {
-  for(unsigned int i=0; i<samples; i++) {
-    int sample = ((int32_t)(uint32_t)(buff[i]) - 0x8000);  // conversion from uint16_t
-    *((uint16_t*)dest) = (int16_t)((sample * simuAudio.currentVolume)/127);
-    dest += 2;
+  for (unsigned i = 0; i < samples; i++) {
+    int16_t sample = buff[i];
+    *((int16_t*)dest) = (int16_t)((sample * simuAudio.currentVolume) / 127);
```

Comment:
potential overflow issue, calculate in int32_t then convert to int16_t with saturation

Reply:
it wouldn't hurt but 
`currentVolume` should be <= 127 and saturation has been already checked before

Reply:
since @raphaelcoeffic chose to use left aligned samples now, the multiplication can overflow

Reply:
in any case, the multiplication has to be done in 32 bit, not 16 bit

Reply:
yes of course

Reply:
I wonder if he really needs cast there, I think no.

Reply:
int32_t sample = buff[i];
*((int16_t*)dest) = _SSAT((sample *  simuAudio.currentVolume) / 127) ,16);